test/unit/utility/bloom_filter/bloom_filter_test.cpp

```diff
@@ -0,0 +1,144 @@
+    EXPECT_TRUE(std::is_copy_assignable_v<TypeParam>);
+    EXPECT_TRUE(std::is_move_assignable_v<TypeParam>);
+    EXPECT_TRUE(std::is_destructible_v<TypeParam>);
+
+    // num hash functions defaults to two
+    TypeParam bf1{TestFixture::make_bf(seqan3::bin_size{1024u})};
+    TypeParam bf2{TestFixture::make_bf(seqan3::bin_size{1024u},
+                                       seqan3::hash_function_count{2u})};
+    EXPECT_TRUE(bf1 == bf2);
+
+     // bin_size parameter is too small
+    EXPECT_THROW((TestFixture::make_bf(seqan3::bin_size{0u})), std::logic_error);
+    // not enough hash functions
+    EXPECT_THROW((TestFixture::make_bf(seqan3::bin_size{32u},
+                                       seqan3::hash_function_count{0u})),
+                 std::logic_error);
+    // too many hash functions
+    EXPECT_THROW((TestFixture::make_bf(seqan3::bin_size{32u},
+                                       seqan3::hash_function_count{6u})),
+                 std::logic_error);
+}
+
+TYPED_TEST(bloom_filter_test, member_getter)
+{
+    TypeParam t1{TestFixture::make_bf(seqan3::bin_size{1024u})};
+    EXPECT_EQ(t1.size(), 1024u);
+    EXPECT_EQ(t1.hash_function_count(), 2u);
+
+    TypeParam t2{TestFixture::make_bf(seqan3::bin_size{1019u},
+                                       seqan3::hash_function_count{3u})};
+    EXPECT_EQ(t2.size(), 1019u);
+    EXPECT_EQ(t2.hash_function_count(), 3u);
+}
+
+TYPED_TEST(bloom_filter_test, contains)
+{
+    TypeParam bf{TestFixture::make_bf(seqan3::bin_size{1024u})};
+
+    for (size_t hash : std::views::iota(0u, 64u)) // test some hashes
+    {
+        // Expect false for all queries since we did not insert anything
+        EXPECT_FALSE(bf.contains(hash)); 
+    }
+}
+
+TYPED_TEST(bloom_filter_test, counting)
+{
+    // 1. Test uncompressed bloom_filter directly because the compressed one is not mutable.
+    seqan3::bloom_filter bf{seqan3::bin_size{1024u},
+                            seqan3::hash_function_count{2u}};
+
+    for (size_t hash : std::views::iota(0u, 128u))
+        bf.emplace(hash);
+
+    // 2. Construct either the uncompressed or compressed bloom_filter and test set with bulk_contains
+    TypeParam bf2{bf};
+    
+    // Test counting with all elements
+    EXPECT_EQ(bf2.count(std::views::iota(0u, 128u)), 128u);
+
+    // Test counting with some elements
+    EXPECT_EQ(bf2.count(std::views::iota(22u, 42u)), 20u);
+}
+
+TYPED_TEST(bloom_filter_test, emplace)
+{
+    // 1. Test uncompressed bloom_filter directly because the compressed one is not mutable.
+    seqan3::bloom_filter bf{seqan3::bin_size{1024u},
+                            seqan3::hash_function_count{2u}};
+
+    for (size_t hash : std::views::iota(0u, 64u))
+        bf.emplace(hash);
+
+    // 2. Construct either the uncompressed or compressed bloom_filter and test via hit counts
+    TypeParam bf2{bf};
+    EXPECT_EQ(bf.count(std::views::iota(0u, 64u)), 64u); // test inserted hashes
+}
```

Comment:
To decouple the tests a bit, I would

* Test `contains` as is
* Test `emplace` and use `contains` to verify
* Test `count` as is

Sorry if I suggested switching to `count` in the `emplace` test before; but otherwise we either assume that `emplace` or `count` just works, and we never explicitly test if `contains` works.